linode/manager
# feat: [M3-7185] - Add Drag and Drop support to the Rules Table

## Description 📝
**Add drag/ drop functionality to Rules Table in Routes Tab**

## Major Changes 🔄
- Added drag and drop functionality to Rules Table.
- Implemented Move up / Move Down action behavior.
- Added useLoadBalancerRouteUpdateMutation.
- Unit tests for RulesTable component.

## Preview 📷
**Include a screenshot or screen recording of the change**
![image](https://github.com/linode/manager/assets/119517080/e4a1406f-2d8e-499c-9b63-6d116c4bc5f6)

## How to test 🧪
1. **How to setup test environment?**
- Check out this PR.
- `yarn dev`
- Ensure AGLB feature flag is **on** and MSW is **enabled**.
2. **How to verify changes?**
- Navigate to http://localhost:3000/loadbalancers/0/routes.
- Click on any route in the table.
- Test Drag and drop functionality in Rules Table.
- Confirm that Move up and Move down actions behaviors.
- Validate Put request integration for updateLoadbalancerRoute according to API spec. 

 
4. **How to run Unit or E2E tests?**
yarn test RulesTable
